ext/sites/stackoverflow.js

```diff
@@ -0,0 +1,72 @@
+const NEW_POST_TIME_CUTOFF_MSEC = 20 * 1000;
+
+const reUserId = /\/users\/([\d\w]+)/;
+const reQuestionPage = /\/questions\/([\d]+)/;
+
+// Only run on question pages
+if (reQuestionPage.test(document.location.pathname)) {
+  const signatures = postSignatures();
+  if (isNewQuestion(signatures)) {
+    stackOverflowMonitor().reason(Reasons.CREATED);
+  }
+  if (isNewAnswer(signatures)) {
+    stackOverflowMonitor().reason(Reasons.COMMENTED);
+  }
+}
+
+function stackOverflowMonitor() {
+  return new Monitor()
+    .pathMatch(reQuestionPage)
+    .past()
+    .sourceID(() => reMatch(document.location.pathname, reQuestionPage, 1))
+    .setProvider('stackoverflow', 'Stack Overflow')
+    .setType(Types.DOCUMENT)
+    .setSubtype(Subtypes.NONE) // TODO: Should we add a sub-type for "Question"?
+    .attachment({
+      name: () => {
+        return document.title.split(' - ')[0];
+      },
+      html_url: () => document.location.href,
+    });
+}
+
+// Returns a parsed array of post signitures.
+// The first post is the original question, the rest are answers.
+// Each post includes: isByCurrentUser and isNew
+function postSignatures() {
+  var sigs = [];
+  // Find the Current UserID from
+  const profile = document.querySelector('.my-profile');
+  const currentUser = reMatch(profile.pathname, reUserId, 1);
+  // Inspect all the question/answer signatures on the page
+  const sigEls = document.querySelectorAll('.post-signature');
+  for (var i = 0; i < sigEls.length; i++) {
+    const userEl = sigEls[i].querySelector('.user-details a');
+    const timeEl = sigEls[i].querySelector('.user-action-time span');
+    if (!userEl || !userEl.pathname || !timeEl || !timeEl.title) break;
```

Comment:
Should this `break` be a `continue`?

Reply:
Yes! Thank you for catching that.

```suggestion
    if (!userEl || !userEl.pathname || !timeEl || !timeEl.title) continue;
``